### PR TITLE
Fix RT685 EVK USDHC support for 1.8V SD Cards

### PR DESCRIPTION
**Pull request by danieldegrasse**
This PR fixes an issue with the MIMXRT685 evaluation board, where SD cards that supported 1.8V signalling would fail to mount with the driver. The core issue was caused because the USDHC driver would recognize that the SD card supported 1.8V signalling, and request the SD card run at that voltage. However, the RT685 EVK uses a PMIC to reduce the signalling voltage on the host side, and the driver did not configure that PMIC correctly. This code adds the correct PMIC configuration and improves the USDHC driver to fall back to 3.3V signalling if 1.8V negotiation fails, so that the RT685 EVK will operate correctly with 1.8V SD cards.

Note: this PR requires an update to the NXP HAL, and should not be merged until the NXP HAL update is accepted

### CODE REVIEW

File: boards/arm/mimxrt685_evk/pinmux.c [hunk 1]
```diff
@@ -961,6 +965,50 @@ static int mimxrt685_evk_pinmux_init(const struct device *dev)
 	IOPCTL_PinMuxSet(IOPCTL, 2U, 9U, port2_pin9_config);
 #endif
 
+#if DT_NODE_HAS_COMPAT_STATUS(DT_NODELABEL(flexcomm15), nxp_lpc_i2c, okay)  && CONFIG_I2C
+	const uint32_t fc15_i2c_scl_config = (/* Pin is configured as I2C_SCL */
+                                          IOPCTL_PIO_FUNC0 |
```
Comment (dleach02) [inline boards/arm/mimxrt685_evk/pinmux.c:970]: need to use <tabs> instead of spaces

File: dts/arm/nxp/nxp_rt6xx_common.dtsi [hunk 2]
```diff
@@ -197,6 +197,24 @@
 		status = "disabled";
 	};
 
+	flexcomm14: flexcomm@126000 {
```
Comment (mmahadevan108) [inline dts/arm/nxp/nxp_rt6xx_common.dtsi:200]: This is already there as `hs_spi`

File: drivers/disk/usdhc.c [hunk 3]
```diff
@@ -12,10 +12,19 @@
 #include <sys/byteorder.h>
 #include <soc.h>
 #include <drivers/clock_control.h>
+#include <fsl_power.h>
+#if CONFIG_SDMMC_USDHC_PMIC_1_8_VOL
+#include <fsl_pca9420.h>
```
Comment (mmahadevan108) [inline drivers/disk/usdhc.c:17]: Do we need this include?

File: drivers/disk/usdhc.c [hunk 4]
```diff
@@ -12,10 +12,19 @@
...
+#endif
 
 #include "sdmmc_sdhc.h"
 
 #include <logging/log.h>
+
+#ifdef CONFIG_BOARD_MIMXRT685_EVK
```
Comment (mmahadevan108) [inline drivers/disk/usdhc.c:24]: Can we use `CONFIG_SDMMC_USDHC_PMIC_1_8_VOL` instead

File: dts/arm/nxp/nxp_rt6xx_common.dtsi [hunk 5]
```diff
@@ -197,6 +197,24 @@
...
+		compatible = "nxp,lpc-spi";
+		reg = <0x126000 0x1000>;
+		interrupts = <20 0>;
+		label = "FLEXCOMM_14";
+		clocks = <&clkctl1 MCUX_FLEXCOMM14_CLK>;
+		status = "disabled";
+	};
+
+	flexcomm15: flexcomm@127000 {
```
Comment (mmahadevan108) [inline dts/arm/nxp/nxp_rt6xx_common.dtsi:209]: Flexcomm 15 is also called PMIC I2C in the reference manual, we could label this a `pmic_i2c`

File: boards/arm/mimxrt685_evk/pinmux.c [hunk 6]
```diff
@@ -1051,5 +1099,84 @@ static int mimxrt685_evk_pinmux_init(const struct device *dev)
 	return 0;
 }
 
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(pca9420), okay) && \
+	CONFIG_SDMMC_USDHC_PMIC_1_8_VOL
```
Comment (mmahadevan108) [inline boards/arm/mimxrt685_evk/pinmux.c:1103]: Can this be a more generic label like `CONFIG_USE_PMIC_PCA9420`

File: boards/arm/mimxrt685_evk/mimxrt685_evk_cm33_defconfig [hunk 7]
```diff
@@ -17,3 +17,6 @@ CONFIG_TRUSTED_EXECUTION_SECURE=y
 
 CONFIG_ARM_MPU=y
 CONFIG_HW_STACK_PROTECTION=y
+
+# Enable PMIC initialization for 1.8V SDHC support
+CONFIG_SDMMC_USDHC_PMIC_1_8_VOL=y
```
Comment (mmahadevan108) [inline boards/arm/mimxrt685_evk/mimxrt685_evk_cm33_defconfig:22]: This should move to the Kconfig.defconfig file with a depends on `SDMMC_USDHC` config

File: boards/arm/mimxrt685_evk/pinmux.c (hunk 6 above)
Comment (danieldegrasse) [reply boards/arm/mimxrt685_evk/pinmux.c:1103]: Yes, do you think we should have 2 KConfig values, or just make the configuration itself more generic?

File: drivers/disk/usdhc.c (hunk 3 above)
Comment (danieldegrasse) [reply drivers/disk/usdhc.c:17]: the fsl_pca9240 is required for the constants kPCA9240_ModeX, but you're right that the fsl_power include is a duplicate

Comment (mmahadevan108) [reply drivers/disk/usdhc.c:16]: My concern is including a board specific include in this driver. It is possible in future we could have a different PMIC on a board. Can we add defines in a board specific header file e.g: `boards/arm/mimxrt685_evk/board.h`

File: boards/arm/mimxrt685_evk/Kconfig.defconfig [hunk 8]
```diff
@@ -40,5 +40,14 @@ endif # DMA_MCUX_LPC
 
 config DISK_DRIVER_SDMMC
 	default y if DISK_DRIVERS
+if SDMMC_USDHC
```
Comment (mmahadevan108) [inline boards/arm/mimxrt685_evk/Kconfig.defconfig:43]: Please use `depends `in the config instead of `if`

**Comment by danieldegrasse** [comment]
Closing PR, since this PR uses the PMIC driver within the HAL and the eventual fix will use a PMIC driver integrated into Zephyr

**Pull request closed without merging**